### ISSUE TITLE
the organize imports operation can change the position of TestOn() annotations

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/sdk#41792]
The organize imports operation can change the position of TestOn() annotations for imports. This changes the semantic meaning of the test annotation - `pub run test` will start running tests on platforms that aren't expected.

See the commit: https://github.com/flutter/devtools/pull/1911/commits/1078d7f325371bca3e2c44f89768f82c4449393d#diff-392849cb2d6b63d0ab4aed6c3162044c for an example. The `@TestOn('vm')` annotation was moved from the beginning of the imports - where package:test will recognize it - to the middle of the imports, where it won't have any effect.

cc @scheglov @bwilkerson

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/146922

**Comment by devoncarew** [comment]
Thanks!

**Comment by scheglov** [comment]
We decided the CL solves  a wrong problem.

**Comment by devoncarew** [comment]
@pq @bwilkerson - did we end up adding the ability to have metadata for annotations? For them to describe where they're legal?

**Comment by pq** [comment]
Brian landed an initial cut of a `Target ` annotation here:

https://github.com/dart-lang/sdk/blob/master/pkg/meta/lib/meta_meta.dart

but it's not (I don't think) published or supported by DAS just yet.

**Comment by bwilkerson** [comment]
Correct. It isn't yet published and there are no hints generated when the contract is violated.

**Comment by devoncarew** [comment]
Great, thanks; I'd love to start consuming it once it's further down the pipeline.
